packages/flutter/test/material/search_anchor_test.dart

```diff
@@ -3233,6 +3234,42 @@ void main() {
 
     expect(suggestionsBuilderCalledCount, 2);
   });
+
+  testWidgets('Suggestions gets refreshed after long API call', (WidgetTester tester) async {
+    Timer? debounceTimer;
+    const Duration apiCallDuration = Duration(seconds: 1);
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        body: Center(
+          child: SearchAnchor(
+            builder: (BuildContext context, SearchController controller) {
+              return const Icon(Icons.search);
+            },
+            suggestionsBuilder: (BuildContext context, SearchController controller) async {
+              final Completer<List<String>> completer = Completer<List<String>>();
+              debounceTimer?.cancel();
+              debounceTimer = Timer(apiCallDuration, () {
+                completer.complete(List<String>.generate(10, (int index) => 'Item - $index'));
+              });
+              final List<String> options = await completer.future;
+
+              final List<Widget> suggestions = List<Widget>.generate(options.length, (int index) {
+                final String item = options[index];
+                return ListTile(
+                  title: Text(item),
+                );
+              });
+              return suggestions;
+            },
+          ),
+        ),
+      ),
+    ));
+    // await tester.pump();
+    await tester.tap(find.byIcon(Icons.search)); // Open search view.
+    await tester.pumpAndSettle(apiCallDuration);
```

Comment:
The `pumpAndSettle` is necessary to wait for all animations to complete. But you are right the test always passed even without the fix because it was missing a part of the use case, which is the Keyboard showing up.
I just fixed it.